core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -207,6 +207,10 @@ package object config {
     .booleanConf
     .createWithDefault(false)
 
+  private[spark] val APP_CALLER_CONTEXT = ConfigBuilder("spark.log.callerContext")
```

Comment:
It shouldn't hurt anything as I don't think there is a hadoop config called that but if there was it would conflict.  In general I think that is a bad idea for us to purposely do this, there is no reason to put it into the hadoop configuration also.  I saw this as something that could apply to other things (then yarn/hdfs) if they supplied the api.  For instance is aws s3 or any other filesystem provided the similar api you could use same config.  But I don't know if it does now.

What is your concern with the current name?